deployment/windows/jellyfin.nsi

```diff
@@ -0,0 +1,202 @@
+; Shows a lot of debug information while compiling
+; This can be removed once stable.
+!verbose 4
+;--------------------------------
+;Include Modern UI
+
+    !include "MUI2.nsh"
+    Var _JELLYFINVERSION_
+    Var _DEFAULTEMBYDATADIR_
+    Var _JELLYFINDATADIR_
+    Var _SERVICEINSTALLED_
+;--------------------------------
+;General
+
+; Align installer version with jellyfin.dll version
+    !getdllversion "$%InstallLocation%\jellyfin.dll" ver_
+    !echo "jellyfin.dll version is ${ver_1}.${ver_2}.${ver_3}.${ver_4}" ;!echo will print it while building
+
+    Name "Jellyfin ${ver_1}.${ver_2}.${ver_3}.${ver_4}"
+    OutFile "Jellyfin.Installer.${ver_1}.${ver_2}.${ver_3}.${ver_4}.exe"
+    BrandingText "Jellyfin ${ver_1}.${ver_2}.${ver_3}.${ver_4} Installer"
+
+; installer attributes
+    VIProductVersion "${ver_1}.${ver_2}.${ver_3}.${ver_4}"
+    VIFileVersion "${ver_1}.${ver_2}.${ver_3}.${ver_4}"
+    VIAddVersionKey "ProductName" "Jellyfin"
+    VIAddVersionKey "FileVersion" "${ver_1}.${ver_2}.${ver_3}.${ver_4}"
+  
+  ;Default installation folder
+    InstallDir "$APPDATA\Jellyfin"
+  
+  ;Get installation folder from registry if available
+    InstallDirRegKey HKLM "Software\Jellyfin" "InstallLocation"
+  
+  ;Request application privileges for Windows Vista
```

Comment:
Should not be needed if installing to AppData (and using HKCU in stead), but I'm not sure if it's possible to request admin on the fly if needed for the chosen install location.

Reply:
You are right, HKCU will not need admin priv. 
I would want to do it this way due to 
1. Ensure install in some protected space (in case the user selects so)
2. Service manipulation (install, remove)
I feel its better to do it upfront rather than midway through install, provides a more consistent user experience.

Reply:
HKLM\Software is the correct location by most guidance given. Ideally we'd avoid the registry entirely, but when you want clean uninstall/install with a configurable location that's not really an option without UWP (which is currently off the table for other reasons.

Reply:
HKLM is for all users install (program files) and HKCU is for appdata installs (current user only) from what I understand. Both are visible in add/remove programs.

Reply:
You're 100% right, but since there is technically nothing stopping someone from installing to Program FIles (and/or running Jellyfin as Local System), the HKLM locations ensures it will be correctly managed. Like there might be a good argument for strongly discouraging that (there are many good arguments for it) but its definitely a configuration that is represented in users (largely because emby used to run as localsystem when it had a service built in).